sdk/storage/storage-blob/src/utils/RetriableReadableStream.ts

```diff
@@ -109,6 +119,12 @@ export class RetriableReadableStream extends Readable {
     this.setSourceDataHandler();
     this.setSourceEndHandler();
     this.setSourceErrorHandler();
+
+    this.source.on("close", () => {
+      if (!this.destroyCalled && !this.aborter.aborted && !this.dataEnded) {
```

Comment:
What if source get into error and received error event?

Reply:
I considered to call its own `destroy()` in every case where the `RetriableReadableStream` emit error. But this changes the old behavior and I am not sure if I should change it. @XiaoningLiu

Reply:
Discussed with Xiaoning and decided to not handle the "close" event as normally there will be an "error" or "end" event before a "close" event.